dao2caom2/telescopes.py

```diff
@@ -0,0 +1,704 @@
+#
+#  You should have received             Vous devriez avoir reçu une
+#  a copy of the GNU Affero             copie de la Licence Générale
+#  General Public License along         Publique GNU Affero avec
+#  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+#  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+#                                       <http://www.gnu.org/licenses/>.
+#
+#  : 4 $
+#
+# ***********************************************************************
+#
+
+import logging
+import re
+
+from caom2 import CalibrationLevel, DataProductType, TargetType, ProductType
+from caom2 import ObservationIntentType
+from caom2pipe import astro_composable as ac
+from caom2pipe import manage_composable as mc
+from dao2caom2.dao_name import  DAOName
+
+
+__all__ = ['factory', 'get_current']
+
+
+class Telescope:
+    """
+    This class is the Spectrum implementation.
+
+    1. OBSMODE keyword => 'imaging' vs 'spectroscopy'
+    2. File name pattern
+
+    All energy is CTYPE = WAVE, CUNIT = A(ngstrom)
+    File Name Pattern
+    all imaging:                       CRVAL    'WAVELENG'
+                                       CDELT    'BANDPASS'
+                                       CRPIX    1
+                                       RP       CRVAL / CDELT
+
+    processed photographic plate spectrum
+    dao_[p]\d{3}_\d{6}(u|v|y|r|i|):
+                                       CRVAL   'CRVAL1'
+                                       CDELT   'CDELT1'
+                                       CRPIX   'CRPIX1'
+                                       RP      CRVAL/(2.5*CDELT)
+
+    unprocessed DAO RETICON spectrum
+    dao_[r]\d{3}_\d{4}_\d{6}:
+                                       CRVAL   ''
+                                       CDELT   ''
+                                       CRPIX   ''
+                                       RP      ''
+
+    unprocessed DAO CCD spectrum
+    dao_[c]\d{3}_\d{4}_\d{6}:
+                                       CRVAL   'WAVELENG'
+                                       CDELT   'DELTA_WL'
+                                       CRPIX   'REFPIXEL'
+                                       RP      CRVAL/(2.5*CDELT)
+
+                                       # DB 04-02-21
+                                       If no 'WAVELENG' no energy
+
+                                       CRVAL   'WAVELENG'
+                                       no `DELTA_WL', 'REFPIXEL'
+                                       CDELT   'DISPERSI' * 15.0 * xbin/1000.0
+                                       CRPIX   'DATASEC' + math
+
+    processed DAO spectrum
+    dao_[cr]\d{3}_\d{4}_\d{6}_[evBF], obs.type in ['object', 'comparison']:
+                                       CRVAL   'CRVAL1'
+                                       CDELT   'CDELT1'
+                                       CRPIX   'CRPIX1'
+                                       RP      CRVAL/(2.5*CDELT)
+
+    """
```

Comment:
BluePrint could not be used with DAO or it was too complicated?

Reply:
I don't understand the comment. BluePrint is used with DAO. This comment just helps me keep clear how the blueprint values are filled in, and why.

Reply:
I don't understand the comment. BluePrint is used with DAO. This comment just helps me keep clear which values are used to fill in the blueprint.

Reply:
The comment was supposed to be for the `accumulate_bp` function and I was just wondering why a config file could not be used instead of those individual sets.

Reply:
That's a good thought. Let me take a look at what kind of changes that would involve.